File: swift/collections/lists.swift
var intArray0: [Int] = []  // Empty array

let immuttableIntArray1 = [5, 4, 6, 533, 32]  // Immutable array
var intArray1 = [1, 2, 4, 5]  // Mutable array

// Most common use caes
intArray1.count  // Returns the number of elements
intArray1.first  // Returns the first element
intArray1.last  // Returns the last element
intArray1.isEmpty  // Returns true if the array is empty
intArray1.filter { $0 > 2 }  // Filters elements greater than 2
intArray1.map { $0 * 2 }  // Doubles each element
intArray1.forEach { print($0) }  // Iterate over the Array and perform action.
intArray1.append(1)  // Adds an element at the end
intArray1.insert(999, at: 1)  // // Add a new element at the index one. one can be any valid index of the Array.
intArray1.remove(at: 0)  // Removes the element at index 0
intArray1.removeLast()  // Remove last element from the Array.

print("Array \(intArray1)")

intArray1.removeAll()  // Clears the array

// Trying add an element on a unmodifiable Array will throw an exception
// immuttableIntArray1.append(190)

File: kotlin/collections/lists.kt
fun main() {

    var intList0: MutableList<Int> = mutableListOf<Int>() // Mutable empty list
    var immutableIntList0: List<Int> = listOf<Int>() // Immutable empty list

    // update values by indices, but won't let add or remove elements.
    var fixedSize: Array<Int> = arrayOf<Int>()

    val noExplicitType1 = mutableListOf(5, 4, 6, 533, 32) // Mutable list

    val intList1: MutableList<Int> = mutableListOf(5, 4, 6, 533, 32) // Mutable list
    val immutableIntList1: List<Int> = listOf(5, 4, 6, 533, 32) // Immutable list

    // Most common use caes
    intList1.size // Returns the number of elements in the list
    intList1.first() // Returns the first element of the list
    intList1.last() // Returns the last element of the list
    intList1.isEmpty() // Returns true if the list is empty
    intList1.isNotEmpty() // Returns true if the list is not empty
    intList1.filter { it > 2 } // Filters elements greater than 2
    intList1.map { it * 2 } // Doubles each element
    intList1.forEach { // Iterate over the list and perform action.
        println(it)
    }
    intList1.add(1) // Adds an element to the end
    intList1.add(
            0,
            1
    ) // Add a new element at the index zero. Zero can be any valid index of the list.
    intList1.removeAt(0) // Removes the element at index 0
    intList1.remove(5) // Remove element 5 from the list. 5 can be any valid element of the list.
    intList1.removeLast() // Removes the last element
    intList1.clear() // Clears the list
}
